Frontend/compile-io-frontend-app/src/app/add-homework/add-homework.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { TestService } from '../services/test.service';
+
+
+@Component({
+  selector: 'app-add-homework',
+  templateUrl: './add-homework.component.html',
+  styleUrls: ['./add-homework.component.css']
+})
+export class AddHomeworkComponent implements OnInit {
+  @Input() username: string;
+  @Input() class: string;
+  file : File;
+
+  constructor(private testService: TestService) {}
+  fileUpload(event: any){
+    console.log("THIS IS THE FILE FROM file upload: " +event.target.files[0]);
+    this.file = event.target.files[0];
+  }
+
+  submit(name: string, time: number, visible: boolean, language: string) {
+    console.log("THIS IS THE FILE FROM SUBMIT: " + this.file);
+    // console.log(name + ", " + time + ", " + visible + ", " + file);
+    // console.log(this.username + "  " + this.class);
+    // console.log(dropDown)
+    this.testService.uploadTest(this.file, language.toLowerCase(), time, name, this.class, this.username).subscribe({
```

Comment:
Just a pointer here for future reference. When making a service call, it should look something more like this.
``` typescript
this.testService.uploadTest(this.file, language.toLowerCase(), time, name, this.class, this.username)
  .pipe(finalize(() => console.log("Uploaded file")))
  .subscribe((x) => {
    console.log(x)
  }, error => {
    console.error(error)
  })
```
The finalize function is a pipable operator imported from rxjs that will get called whether you have an error or not.